src/SFA.DAS.EmployerAccounts/Services/UserAornPayeLockService.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using SFA.DAS.EmployerAccounts.Configuration;
+using SFA.DAS.EmployerAccounts.Interfaces;
+using SFA.DAS.EmployerAccounts.Models.UserProfile;
+using SFA.DAS.NLog.Logger;
+
+namespace SFA.DAS.EmployerAccounts.Services
+{
+    public class UserAornPayeLockService : IUserAornPayeLockService
+    {
+        private readonly IUserRepository _userRepository;
+        private readonly UserAornPayeLockConfiguration _configuration;
+        private readonly ILog _logger;
+
+        public UserAornPayeLockService(
+            IUserRepository userRepository, 
+            EmployerAccountsConfiguration configuration,
+            ILog logger)
+        {
+            _logger = logger;
+            _userRepository = userRepository;
+            _configuration = configuration.UserAornPayeLock;
+        }
+
+        public async Task<bool> UpdateUserAornPayeAttempt(Guid userRef, bool success)
+        {
+            try
+            {
+                await _userRepository.UpdateAornPayeQueryAttempt(userRef, success);
+            }
+            catch (Exception e)
+            {
+                _logger.Error(e, "An error occurred when calling the Aorn Lock Service.");
+                return false;
+            }
+            
+            return true;
+        }
+
+        public async Task<UserAornPayeStatus> UserAornPayeStatus(Guid userRef)

```

Comment:
This method could do with a quick refactor to extract a few bits out to make it more readable.